jspackage/jspackage.go

```diff
@@ -10,30 +10,28 @@ import (
 	"github.com/bitrise-io/go-utils/pathutil"
 )
 
-// JsPackageManager describes wether npm or yarn is used
-type JsPackageManager int
+// Manager identifies a package manager tool
+type Manager string
 
-// JsPackageManager types
+// Package manager types
 const (
-	Npm  JsPackageManager = 0
-	Yarn JsPackageManager = 1
+	Npm  Manager = "npm"
+	Yarn Manager = "yarn"
 )
 
-func detectJsPackageManager(absPackageJSONDir string) JsPackageManager {
+// DetectManager returns the Js package manager used, e.g. npm or yarn
+func DetectManager(absPackageJSONDir string) Manager {
 	if exist, err := pathutil.IsPathExists(filepath.Join(absPackageJSONDir, "yarn.lock")); err != nil {
 		log.Warnf("Failed to check if yarn.lock file exists in the workdir: %s", err)
-		log.TPrintf("Package manager: npm")
 		return Npm
 	} else if exist {
-		log.TPrintf("Package manager: yarn")
 		return Yarn
-	} else {
-		log.TPrintf("Package manager: npm")
-		return Npm
 	}
+	return Npm
 }
 
-func removeJsPackages(packageManager JsPackageManager, isGlobal bool, pkg ...string) error {
+// Remove removes installed js dependencies using the selected package manager
+func Remove(packageManager Manager, isGlobal bool, pkg ...string) error {
```

Comment:
Remove and Add seems to be really similar, may it would worth to separate a new function for the common parts (everything is common expect to remove/install and remove/add)